src/FocusApp.Client/Views/Social/IslandsPage.cs

```diff
@@ -0,0 +1,327 @@
+				new Button
+                {
+                    Text = SolidIcons.ChevronLeft,
+                    TextColor = Colors.Black,
+                    FontFamily = nameof(SolidIcons),
+                    FontSize = 40,
+                    BackgroundColor = Colors.Transparent
+                }
+                .Left()
+                .CenterVertical()
+                .Column(0)
+                .Paddings(top : 5, bottom: 5, left: 10, right: 5)
+				// When clicked, go to social view
+				.Invoke(button => button.Released += (sender, eventArgs) =>
+                    BackButtonClicked(sender, eventArgs)),
+
+				// Header & Content Divider
+				new BoxView
+                {
+                    Color = Colors.Black,
+                    WidthRequest = 400,
+                    HeightRequest = 2
+                }
+                .Bottom()
+                .Row(0)
+                .Column(0)
+                .ColumnSpan(int.MaxValue),
+                //////////////////////////////////////////////////////////
+
+				// FlexLayout - Container for Islands
+                new ScrollView
+                {
+                    Content = _islandsContainer
+                }
+                .Row(2)
+                .Column(0)
+                .ColumnSpan(2)
+                .Top()
+                .Paddings(bottom: 10),
+
+                // Label for response message
+                _responseMessage
+                .Row(3)
+                .Column(0)
+                .ColumnSpan(2)
+                .Center()
+            }
+		};
+	}
+
+    protected override async void OnAppearing()
+    {
+        base.OnAppearing();
+
+        // Check if the user is logged in
+        if (_authenticationService.CurrentUser == null)
+        {
+            var loginPopup = (EnsureLoginPopupInterface)_popupService.ShowAndGetPopup<EnsureLoginPopupInterface>();
+            loginPopup.OriginPage = nameof(ShopPage);
+            return;
+        }
+
+        // Clear message label
+        _responseMessage.Text = "";
+
+        // Fetch islands from the local database and display them
+        var userIslands = await FetchIslandsFromLocalDb();
+        DisplayIslands(userIslands);
+    }
+
+    private async Task<List<IslandItem>> FetchIslandsFromLocalDb()
+    {
+        List<IslandItem> userIslands = new List<IslandItem>();
+
+        try
+        {
+            // Fetch islands from the local database using Mediatr feature
+            GetUserIslands.Result result = await _mediator.Send(new GetUserIslands.Query()
+            {
+                UserId = _authenticationService.CurrentUser.Id,
+                selectedIslandId = _authenticationService.SelectedIsland != null ? _authenticationService.SelectedIsland.Id : null
+            },
+            default);
+
+            userIslands = result.Islands;
+        }
+        catch (Exception ex)
+        {
+            throw new Exception("Error when fetching islands from local DB.", ex);
```

Comment:
Can we change this to be logging? I think throwing an uncaught exception might crash the app

Reply:
If this happens on the pets and decor pages, then we need to do the same there as well

Reply:
on it!